app/src/main/java/io/intrepid/contest/screens/contestcreation/addcategoriestocontest/AddCategoriesPresenter.java

```diff
@@ -0,0 +1,48 @@
+package io.intrepid.contest.screens.contestcreation.addcategoriestocontest;
+
+import android.support.annotation.NonNull;
+
+import java.util.List;
+
+import io.intrepid.contest.base.BasePresenter;
+import io.intrepid.contest.base.PresenterConfiguration;
+import io.intrepid.contest.models.Category;
+import io.intrepid.contest.models.Contest;
+import io.intrepid.contest.models.ContestProvider;
+
+import static io.intrepid.contest.screens.contestcreation.addcategoriestocontest.AddCategoriesContract.View;
+
+
+class AddCategoriesPresenter extends BasePresenter<AddCategoriesContract.View> implements AddCategoriesContract.Presenter {
+    private List<Category> categories;
+    private Category editableCategory;
+
+    AddCategoriesPresenter(@NonNull View view, @NonNull PresenterConfiguration configuration) {
+        super(view, configuration);
+        categories = ContestProvider.getInstance().getLastEditedContest().categories;
+        editableCategory = new Category();
```

Comment:
I discussed another approach to this with @jmccormack200 . Will send off event callbacks from the different fragments to the activity directly instead. Your comment makes sense too. If I'm doing this with interface callbacks, the active contest being edited can be a parameter in one of the interface methods. It's a lot more intuitive than my current approach